great_expectations/data_context/templates.py

```diff
@@ -1,66 +1,63 @@
-# datasource or generator, be sure to also update the paths for related artifacts.
+# NOTE: GE uses the names of configured `datasources` and `generators` to manage
+# how `expectations` and other artifacts are stored in the `expectations/` and 
+# `datasources/` folders. If you need to rename an existing `datasource` or 
+# `generator`, be sure to also update the relevant directory names.
 
-ge_config_version: 1
+config_version: 1
 
-"""
+""".format(rtd_url_ge_version)
 
 CONFIG_VARIABLES_INTRO = """
-# Great Expectations config file supports variable substitution.
-
-# Variable substitution enables these two use cases:
-# 1. do not store sensitive credentials in a committed file (since the credentials file should be in 
-#    the uncommitted directory  
-# 2. allow a config parameter to take different values, depending on the environment (e.g., dev/staging/prod)
-#
-# When GE encounters the following syntax in the config file:
-#
-# my_key: ${my_value} (or $my_value)
+# This config file supports variable substitution which enables two use cases:
+#   1. Secrets are kept out of committed files.
+#   2. Configuration parameters can change based on the environment. For
+#      example: dev vs staging vs prod.
 #
-# GE will attempt to replace the value of “my_key” with the value of env variable “my_value” or with the value of the key “my_value” read from credentials file (env variable takes precedence).
+# When GE encounters substitution syntaxes in the config file, like the ones
+# below it will attempt to replace the value of “my_key” with the value from an 
+# environment variable “my_value” or a corresponding key read from the file
+# specified using `config_variables_file_path`. An environment variable will
+# always take precedence.
+#   - `my_key: ${my_value}
+#   - `my_key: $my_value`
 #
-# If the replacing value comes from the config variables file, it can be a simple value or a dictionary. If it comes from an environment variable, it must be a simple value.
-"""
+# If the substitution value comes from the config variables file, it can be a
+# simple (non-nested) value or a nested value such as a dictionary. If it comes
+# from an environment variable, it must be a simple value."""
 
 PROJECT_OPTIONAL_CONFIG_COMMENT = CONFIG_VARIABLES_INTRO + """
 config_variables_file_path: uncommitted/config_variables.yml
 
-# The plugins_directory is where the data_context will look for custom_data_assets.py
-# and any configured evaluation parameter store
-
+# The plugins_directory will be added to your python path for custom modules
+# used to override and extend Great Expectations.
 plugins_directory: plugins/
 
 expectations_store:
+  # This is where expectations are kept.
   class_name: ExpectationStore
   store_backend:
     class_name: FixedLengthTupleFilesystemStoreBackend
     base_directory: expectations/
 
 profiling_store_name: local_validation_result_store
-
 evaluation_parameter_store_name: evaluation_parameter_store
```

Comment:
I think all required stores should be somehow called out, and all should either use ..._name or should be defined in place, but not a mix (as here)

Reply:
This is now addressed in https://github.com/great-expectations/great_expectations/pull/745